benchmarks/src/main/java/io/grpc/benchmarks/driver/LoadClient.java

```diff
@@ -0,0 +1,522 @@
+        PooledByteBufAllocator alloc = PooledByteBufAllocator.DEFAULT;
+        genericRequest = alloc.buffer(config.getPayloadConfig().getBytebufParams().getRespSize());
+        if (genericRequest.capacity() > 0) {
+          genericRequest.writerIndex(genericRequest.capacity() - 1);
+        }
+        break;
+      }
+      default: {
+        // Not implemented yet
+        throw new IllegalArgumentException("Scenario not implemented");
+      }
+    }
+
+    List<OperatingSystemMXBean> beans =
+        ManagementFactory.getPlatformMXBeans(OperatingSystemMXBean.class);
+    if (!beans.isEmpty()) {
+      osBean = beans.get(0);
+    } else {
+      osBean = null;
+    }
+
+    // Create the histogram recorder
+    recorder = new Recorder((long) config.getHistogramParams().getMaxPossible(), 3);
+  }
+
+  /**
+   * Start the load scenario.
+   */
+  void start() {
+    Runnable r;
+    for (int i = 0; i < threadCount; i++) {
+      r = null;
+      switch (config.getPayloadConfig().getPayloadCase()) {
+        case SIMPLE_PARAMS: {
+          if (config.getClientType() == Control.ClientType.SYNC_CLIENT) {
+            if (config.getRpcType() == Control.RpcType.UNARY) {
+              r = new BlockingUnaryWorker(blockingStubs[i % blockingStubs.length]);
+            }
+          } else if (config.getClientType() == Control.ClientType.ASYNC_CLIENT) {
+            if (config.getRpcType() == Control.RpcType.UNARY) {
+              r = new AsyncUnaryWorker(asyncStubs[i % asyncStubs.length]);
+            } else if (config.getRpcType() == Control.RpcType.STREAMING) {
+              r = new AsyncPingPongWorker(asyncStubs[i % asyncStubs.length]);
+            }
+          }
+          break;
+        }
+        case BYTEBUF_PARAMS: {
+          if (config.getClientType() == Control.ClientType.SYNC_CLIENT) {
+            if (config.getRpcType() == Control.RpcType.UNARY) {
+              r = new GenericBlockingUnaryWorker(channels[i % channels.length]);
+            }
+          } else if (config.getClientType() == Control.ClientType.ASYNC_CLIENT) {
+            if (config.getRpcType() == Control.RpcType.UNARY) {
+              r = new GenericAsyncUnaryWorker(channels[i % channels.length]);
+            } else if (config.getRpcType() == Control.RpcType.STREAMING) {
+              r = new GenericAsyncPingPongWorker(channels[i % channels.length]);
+            }
+          }
+
+          break;
+        }
+        default: {
+          throw Status.UNIMPLEMENTED.asRuntimeException();
+        }
+      }
+      if (r == null) {
+        throw new IllegalStateException(config.getRpcType().name()
+            + " not supported for client type "
+            + config.getClientType());
+      }
+      fixedThreadPool.execute(r);
+    }
+    if (osBean != null) {
+      lastMarkCpuTime = osBean.getProcessCpuTime();
+    }
+  }
+
+  /**
+   * Take a snapshot of the statistics which can be returned to the driver.
+   */
+  Stats.ClientStats getStats() {
```

Comment:
can we make this testable? Pass in a histogram and add a unit test that verifies that conversion from Histogram to ClientStats is correct or so? WDYT?